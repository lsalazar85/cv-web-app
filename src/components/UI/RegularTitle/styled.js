import styled from 'styled-components'

const H2 = styled.h2`
    color: #ffffff;
    font-weight: 700;
    font-size: ${props => `calc(${props.minFontSize}px + (${props.maxFontSize} - ${props.minFontSize}) * ((100vw - 300px) / (1600 - 300)))`}
`

export { H2 };