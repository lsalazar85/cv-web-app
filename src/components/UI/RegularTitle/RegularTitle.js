import React from 'react';
import { string } from  'prop-types';

import { H2 } from './styled';

const RegularTitle = ({ content, minFontSize, maxFontSize }) => (
    <div><H2 minFontSize={minFontSize} maxFontSize={maxFontSize}>{content}</H2></div>
)

RegularTitle.propTypes = {
    content: string.isRequired,
    minFontSize: string.isRequired,
    maxFontSize: string.isRequired,
}


export default RegularTitle;