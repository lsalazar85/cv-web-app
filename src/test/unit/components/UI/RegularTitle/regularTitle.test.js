import React from 'react';
import { render, cleanup } from '@testing-library/react';

import RegularTitle from '../../../../../components/UI/RegularTitle';

describe('RegularTitle', () => {
    afterEach(cleanup);

    it('should render correctly', () => {
        const { asFragment } = render(
            <RegularTitle content={'Luis Salazar'} maxFontSize={'48'} minFontSize={'18'} />
        );
        expect(asFragment()).toBeDefined();
        expect(asFragment()).toMatchSnapshot();
    });
})