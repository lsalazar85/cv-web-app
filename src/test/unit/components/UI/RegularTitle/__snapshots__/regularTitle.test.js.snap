// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegularTitle should render correctly 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="sc-bdnylx cVLySl"
    >
      Luis Salazar
    </h2>
  </div>
</DocumentFragment>
`;
